src/components/providers/PushNotificationsProvider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { Alert, Platform } from "react-native";
+import * as Notifications from "expo-notifications";
+import * as Device from "expo-device";
+
+type PushNotificationsContextState = {
+  isPushEnabled: boolean;
+  setPushEnabled?: (value: boolean) => void;
+  sendPushNotification?: (notificationContent: {
+    title: string;
+    body: string;
+  }) => Promise<void>;
+};
+
+const initialState: PushNotificationsContextState = {
+  isPushEnabled: false,
+};
+
```

Comment:
Maybe we can handle the both in-app and push notifications in the same provider, we could rename this more generic like NotificationsContext and so the rest,  what do you think